packages/gestalt/src/PageHeader.js

```diff
@@ -0,0 +1,72 @@
+// @flow strict
+import React, { type Node, Children } from 'react';
+import Box from './Box.js';
+import Flex from './Flex.js';
+import Heading from './Heading.js';
+import Text from './Text.js';
+import { type Dimension } from './boxTypes.js';
+
+type Props = {|
+  title: string,
+  maxWidth?: Dimension,
+  primaryAction?: Node,
+  secondaryAction?: Node,
+  subtext?: string,
+|};
+
+export default function PageHeader({
+  maxWidth = '100%',
+  primaryAction,
+  secondaryAction,
+  subtext,
+  title,
+}: Props): Node {
+  const renderAction = (action?: Node): Node => {
+    // This is technically all temporary, until we create a proper Menu component
+    // Then, we could type check using Flow for Buttons, Links, or Menus
+    let allowedChildFound = false;
+    if (
+      action.type?.displayName &&
+      ['IconButton', 'Button', 'Link', 'Tooltip'].includes(action.type.displayName)
+    ) {
+      allowedChildFound = true;
+    } else if (action.props?.children) {
+      const actionChildrenArray = Children.toArray(action.props.children);
+      actionChildrenArray.forEach((child) => {
+        if (
+          child.type &&
+          ['IconButton', 'Button', 'Link', 'Tooltip'].includes(child.type.displayName)
+        ) {
+          allowedChildFound = true;
+        }
+      });
+    }
+    if (!allowedChildFound) {
+      throw new Error('PageHeader actions must be of type Button, IconButton, Link, or Tooltip.');
+    }
+
+    return allowedChildFound ? action : <div />;
```

Comment:
I would think we'd want to return null  here if the action is invalid? We don't want random divs for no reason